src/extension.ts

```diff
@@ -147,22 +145,37 @@ class RunLineInTerminal {
     }
 }
 
+const elegantSpinner = require('elegant-spinner');
+
 class RunLineInEditor {
 
     private resultDocument: TextDocument | undefined;
     private parsedResult: object | undefined;
     private queryEnabled = false;
     private query: string | undefined;
     private disposables: Disposable[] = [];
+    private runStatusBarItem: StatusBarItem;
+    private interval!: NodeJS.Timer;
+    private spinner = elegantSpinner();
+    private hideStatusBarItemTimeout! : NodeJS.Timeout;
 
     constructor(private status: StatusBarInfo) {
         this.disposables.push(commands.registerTextEditorCommand('ms-azurecli.toggleLiveQuery', editor => this.toggleQuery(editor)));
         this.disposables.push(commands.registerTextEditorCommand('ms-azurecli.runLineInEditor', editor => this.run(editor)));
         this.disposables.push(workspace.onDidCloseTextDocument(document => this.close(document)));
         this.disposables.push(workspace.onDidChangeTextDocument(event => this.change(event)));
+
+        this.runStatusBarItem = window.createStatusBarItem(StatusBarAlignment.Left);
     }
 
     private run(source: TextEditor) {
+        const t0 = Date.now();
+        this.interval = setInterval(() => {
```

Comment:
This does not yet handle the case where the user executes a second command before the first has finished.

Reply:
Added some code to support commands running simultaneously. Keeping count of how many commands are running (runningCommandCount) feels a little kludgy to me but I all the alternatives I tried were worse.